syne_tune/backend/sagemaker_backend/sagemaker_backend.py

```diff
@@ -194,9 +203,62 @@ def _checkpoint_s3_uri_for_trial(self, trial_id: int) -> str:
             res_path = f"{res_path}/{self.tuner_name}"
         return f"{res_path}/{str(trial_id)}/checkpoints/"
 
+    def _config_json_filename(self, trial_id: int, with_path: bool) -> str:
+        fname = CONFIG_JSON_FILENAME + f"_{trial_id}.json"
+        if with_path and self.source_dir is not None:
+            return str(Path(self.source_dir) / fname)
+        else:
+            return fname
+
+    def _hyperparameters_from_config(
```

Comment:
We should split this function into two as it accomplishes two things: 1/ translate config into hyperparams and 2/ store config into json if `self.pass_args_as_json`. We should have two functions a/ `_hyperparameters_from_config` which just does the translation and b/ `_store_config_in_json` which will take care of the writing. We can call them from `_schedule` as 

```
hyperparameters = self._hyperparameters_from_config(trial_id, config)
if self.pass_args_as_json:
    self._store_config_in_json(trial_id, config)
```

Reply:
OK, I split it into two. However, both 1/ and 2/ are only non-trivial (and connected) if `self.pass_args_as_json`.

Basically, the point is distributing the config between CL arguments (i.e., hyperparameters returned) and what is written as JSON.

It is simpler this way, i.e. passing default args with names "ST_..." *always* on the command line, because there are other helper functions for parsing these args. These args always have simple types (int, float, str), so it does not matter